packages/doenetml-prototype/src/renderers/doenet/graph.tsx

```diff
@@ -98,9 +119,70 @@ export const Graph: BasicComponent = ({ node }) => {
         );
         setYaxis(yaxis);
 
+        board.on("boundingbox", () => {
+            if (!settingBoundingBox.current) {
+                let newBoundingBox = board.getBoundingBox();
+                let [xMin, yMax, xMax, yMin] = newBoundingBox;
+
+                // look for a change in bounding box that isn't due to round-off error
+                let xScale = Math.abs(xMax - xMin);
+                let yScale = Math.abs(yMax - yMin);
+                let diffs = newBoundingBox.map((v, i) =>
+                    Math.abs(v - (previousBoundingBox.current?.[i] ?? 0)),
+                );
+                if (
+                    Math.max(
+                        diffs[0] / xScale,
+                        diffs[1] / yScale,
+                        diffs[2] / xScale,
+                        diffs[3] / yScale,
+                    ) > 1e-12
+                ) {
+                    previousBoundingBox.current = newBoundingBox;
+
+                    let action: Action = {
+                        component: "graph",
+                        actionName: "changeAxisLimits",
+                        componentIdx: node.data.id,
+                        args: {
+                            x_min: xMin,
+                            x_max: xMax,
+                            y_min: yMin,
+                            y_max: yMax,
+                        },
+                    };
+                    dispatch(coreActions.dispatchAction(action));
+                }
+            }
+        });
+
         setBoard(board);
+
+        return () => {
+            board.off("boundingbox");
+        };
     }, [boardRef]);
 
+    let boundingBox: BoundingBox = [
+        node.data.props.xMin,
+        node.data.props.yMax,
+        node.data.props.xMax,
+        node.data.props.yMin,
+    ];
+
+    if (
+        board &&
+        boundingBox.some((v, i) => v !== previousBoundingBox.current?.[i])
```

Comment:
I'd just make a util `function arrayEq` and use that so the purpose is clearer. (I'm sure one of the libraries we have already includes such a thing, but it's such a small function it couldn't hurt)